src/components/legend/style.js

```diff
@@ -145,7 +145,7 @@ module.exports = function style(s, gd, legend) {
 
         // with fill and no markers or text, move the line and fill up a bit
         // so it's more centered
-        var markersOrText = subTypes.hasMarkers(trace) || subTypes.hasText(trace);
+        var markersOrText = subTypes.hasMarkers(trace) || subTypes.isText(trace);
```

Comment:
I believe this change is correct, though I don't see any test images in which the legend symbol shifted position due to the text being removed. Probably OK, just want to note it as a gap. This would need to be something like `mode: 'lines+text', fill: 'tozeroy'`.

Actually this raises a question: what if we have just fill and text, `mode: 'text', fill: 'tozeroy'` - do we still want to show the `Aa` in that case, or only the fill? Currently in this PR we would still show `Aa` but I might argue for dropping it, ie showing text only when there's nothing else at all in the legend. @nicolaskruchten thoughts?

Reply:
Yeah, let's make this rule "if rendering no text would result in no legend item at all, show the text". So if fills without lines cause a marker to appear, then no text please.

Reply:
Addressed in https://github.com/plotly/plotly.js/pull/5682/commits/10d1138160868415c1fa3dc633c592ce8b03be2d.